.buildkite/scripts/lucene-snapshot/update-es-snapshot.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -euo pipefail
+
+echo --- Update Lucene snapshot in Elasticsearch
+
+LUCENE_SNAPSHOT_VERSION=${LUCENE_SNAPSHOT_VERSION:-}
+
+if [[ -z "$LUCENE_SNAPSHOT_VERSION" ]]; then
+  LUCENE_SNAPSHOT_VERSION=$(buildkite-agent meta-data get lucene-snapshot-version)
+fi
+
+echo "Lucene Snapshot Version: $LUCENE_SNAPSHOT_VERSION"
+
+git checkout "$BUILDKITE_BRANCH"
+git pull --ff-only origin "$BUILDKITE_BRANCH"
+
+# Replace `lucene = <version>` string in version.properties and maintain the same indentation
+sed -E "s/^(lucene *= *[^ ]*  *).*\$/\1$LUCENE_SNAPSHOT_VERSION/" build-tools-internal/version.properties > new-version.properties
+mv new-version.properties build-tools-internal/version.properties
+
+# Remove stale verification metadata, because generating them just appends new ones
+python3 .buildkite/scripts/lucene-snapshot/remove-verification-metadata.py
+./gradlew --write-verification-metadata sha256
+
+# 9.8.0-snapshot-1f25c68 -> 9.8.0
+VERSION=$(echo "$LUCENE_SNAPSHOT_VERSION" | cut -f 1 -d '-')
+VERSION_SNAKE=$(echo "$VERSION" | sed -E 's/\./_/g')
+
+sed -E "s/^(:lucene_version:  *).*\$/\1$VERSION/" docs/Versions.asciidoc > docs/Versions.asciidoc.new
```

Comment:
I wonder if this is really even necessary, given we'll never release with a snapshot version, and this can just be done whenever we bump main to the final release version.

cc @jpountz Thoughts?

Reply:
I added it because there are checks that fail if it doesn't match.